### PR TITLE
EPROD-259 register-agent cli

### PR DESCRIPTION
**Pull request by veeso**
# Account registration CLI tool

Issue ticket link: <https://infinityswap.atlassian.net/jira/software/c/projects/EPROD/boards/16?modal=detail&selectedIssue=EPROD-259>

### Code conventions

- [x] I have performed a self-review of my code
- [x] Every new function is documented
- [x] Object names are auto explicative
- [x] The PR follows the [project conventions](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/23330839/Conventions)
- [x] The code follows the [style guidelines of the project](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/24444929/Rust+Conventions)
- [x] The code follows the [project security best practices](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/35225620/Security+Considerations)

### Security

- [x] The PR does not break APIs backward compatibility
- [x] The PR does not break the stable storage backward compatibility

### Testing

- [ ] Every function is properly unit tested
- [ ] I have added integration tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

**Comment by veeso** [comment]
blocked by https://infinityswap.atlassian.net/jira/software/c/projects/EPROD/boards/16?modal=detail&selectedIssue=EPROD-374

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Issue ticket link: <>
+
+## Checklist before requesting a review
+
+### Code conventions
+
+- [ ] I have performed a self-review of my code
+- [ ] Every new function is documented
+- [ ] Object names are auto explicative
+- [ ] The PR follows the [project conventions](https://infinityswap.atlassian.net/wiki/spaces/CPROD/pages/23330839/Conventions)
```
Comment (ufoscout) [inline .github/pull_request_template.md:12]: Please remove links to the private pages in `infinityswap.atlassian.net/wiki`

File: Cargo.toml [hunk 2]
```diff
@@ -10,25 +9,26 @@ edition = "2021"
 [workspace.dependencies]
 anyhow = "1.0"
 candid = "0.8"
+clap = { version = "4.2.4", features = ["derive"] }
+dirs = "5.0"
+env_logger = "^0.10"
+eth-signer = { git = "https://github.com/infinity-swap/eth-signer" }
```
Comment (ufoscout) [inline Cargo.toml:15]: use a tagged version

File: Cargo.toml [hunk 3]
```diff
@@ -10,25 +9,26 @@ edition = "2021"
...
 # infinityswap.atlassian.net/browse/EPROD-153 - We are currently depending on a commit, will be removed when new version is released
 ethers-core = { git = "https://github.com/infinity-swap/ethers-rs", default-features = false, package = "ethers-core", branch = "evmc_fork" }
-serde_bytes = "0.11"
-clap = { version = "4.2.4", features = ["derive"] }
+evmc-did = { git = "https://github.com/infinity-swap/evmc-did" }
```
Comment (ufoscout) [inline Cargo.toml:18]: use a tagged version

File: src/register_evm_agent/src/agent/generic_identity.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+pub enum GenericIdentity {
+    Secp256k1Identity(Secp256k1Identity),
+    BasicIdentity(BasicIdentity),
+}
+
+impl TryFrom<&Path> for GenericIdentity {
+    type Error = Error;
+
+    fn try_from(path: &Path) -> Result<Self, Self::Error> {
```
Comment (ufoscout) [inline src/register_evm_agent/src/agent/generic_identity.rs:17]: please add a test

File: src/register_evm_agent/src/agent/generic_identity.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        Secp256k1Identity::from_pem_file(path)
+            .map(GenericIdentity::from)
+            .or(BasicIdentity::from_pem_file(path).map(GenericIdentity::from))
+            .map_err(|e| Error::Pem(path.to_path_buf(), e))
+    }
+}
+
+impl Identity for GenericIdentity {
+    fn sender(&self) -> std::result::Result<Principal, String> {
```
Comment (ufoscout) [inline src/register_evm_agent/src/agent/generic_identity.rs:26]: add a test

File: src/register_evm_agent/src/agent/generic_identity.rs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        match self {
+            Self::BasicIdentity(identity) => identity.sender(),
+            Self::Secp256k1Identity(identity) => identity.sender(),
+        }
+    }
+
+    fn sign(&self, blob: &[u8]) -> std::result::Result<ic_agent::Signature, String> {
```
Comment (ufoscout) [inline src/register_evm_agent/src/agent/generic_identity.rs:33]: add a test

**Review by ufoscout — CHANGES_REQUESTED** [review]

**Pull request merged**